resource_container/ResourceContainer.py

```diff
@@ -0,0 +1,623 @@
+        :return string: 
+        """
+        return {
+            'text/usx': 'usx',
+            'text/usfm': 'usfm',
+            'text/markdown': 'md'
+        }.get(self.format, 'txt')
+
+    @property
+    def type(self):
+        if 'type' in self.resource and isinstance(self.resource['type'], basestring):
+            return self.resource['type']
+        elif self.identifier in ['udb', 'ulb', 'bible']:
+            if len(self.rc.usfm_files()):
+                return 'bundle'
+            else:
+                return resource_map[self.identifier]['type']
+        elif self.identifier in resource_map:
+            return resource_map[self.identifier]['type']
+        else:
+            return 'book'
+
+    @property
+    def identifier(self):
+        if 'identifier' in self.resource and self.resource['identifier']:
+            return self.resource['identifier']
+        if 'id' in self.resource and self.resource['id']:
+            return self.resource['id']
+        elif 'slug' in self.resource and self.resource['slug']:
+            slug = self.resource['slug']
+            if 'ulb' in slug:
+                return 'ulb'
+            elif 'udb' in slug:
+                return 'udb'
+            elif 'obs' in slug:
+                return 'obs'
+            else:
+                return slug
+        elif 'ulb' in self.rc.repo_name:
+            return 'ulb'
+        elif 'udb' in self.rc.repo_name:
+            return 'udb'
+        elif 'obs' in self.rc.repo_name:
+            return 'obs'
+        else:
+            return 'bible'
+
+    @property
+    def title(self):
+        if 'title' in self.resource and self.resource['title']:
+            return self.resource['title']
+        elif 'name' in self.resource and self.resource['name']:
+            return self.resource['name']
+        elif self.identifier in resource_map:
+            return resource_map[self.identifier]['title']
+        else:
+            return self.identifier
+
+    @property
+    def subject(self):
+        return self.resource.get('subject', self.title)
+
+    @property
+    def description(self):
+        return self.resource.get('description', self.title)
+
+    @property
+    def relation(self):
+        return self.resource.get('relation', [])
+
+    @property
+    def publisher(self):
+        return self.resource.get('publisher', 'Door43')
+
+    @property
+    def issued(self):
+        if 'issued' in self.resource and self.resource['issued']:
+            return self.resource.get('issued')
+        elif 'pub_date' in self.resource.get('status', {}):
+            return self.resource['status']['pub_date']
+        else:
+            return datetime.utcnow().strftime("%Y-%m-%d")
+
+    @property
+    def modified(self):
+        return self.resource.get('modified', datetime.utcnow().strftime("%Y-%m-%d"))
+
+    @property
+    def rights(self):
+        return self.resource.get('rights', 'CC BY-SA')
```

Comment:
Ok, will update. Not sure which manifest example I got that from. Maybe the readthedocs?